sql/moz-fx-cjms-nonprod-9a36/cjms_bigquery/subscriptions_v1/query.sql

```diff
@@ -0,0 +1,72 @@
+WITH flows AS (
+  -- last fxa_uid for each flow_id
+  SELECT
+    flow_id,
+    MIN(flow_started) AS flow_started,
+    ARRAY_AGG(fxa_uid IGNORE NULLS ORDER BY fxa_uid_timestamp DESC LIMIT 1)[
+      SAFE_OFFSET(0)
+    ] AS fxa_uid,
+  FROM
+    `moz-fx-cjms-nonprod-9a36`.cjms_bigquery.flows_v1
+  WHERE
+    -- only use the last 10 days in stage
+    submission_date >= CURRENT_DATE - 10
+  GROUP BY
+    flow_id
+),
+attributed_flows AS (
+  -- last flow_id in aic for each fxa_uid
+  SELECT
+    fxa_uid,
+    ARRAY_AGG(STRUCT(flow_id, flow_started) ORDER BY flow_started DESC LIMIT 1)[SAFE_OFFSET(0)].*,
+  FROM
+    flows
+  JOIN
+    EXTERNAL_QUERY("moz-fx-cjms-nonprod-9a36.us.cjms-sql", "SELECT flow_id FROM aic")
+  USING
+    (flow_id)
+  GROUP BY
+    fxa_uid
+),
+attributed_subs AS (
+  -- last subscription_id created after the flow started for each fxa_uid
+  SELECT
+    attributed_flows.fxa_uid,
+    ARRAY_AGG(
+      nonprod_stripe_subscriptions.subscription_id
+      ORDER BY
+        nonprod_stripe_subscriptions.created DESC
+      LIMIT
+        1
+    )[SAFE_OFFSET(0)] AS subscription_id,
+  FROM
+    attributed_flows
+  JOIN
+    mozdata.subscription_platform.nonprod_stripe_subscriptions
+  ON
+    attributed_flows.fxa_uid = nonprod_stripe_subscriptions.fxa_uid
+    AND attributed_flows.flow_started < nonprod_stripe_subscriptions.created
+  GROUP BY
+    fxa_uid
+)
```

Comment:
- This seems backward to me, as I'd think you'd want the last attributed flow for the FxA before the subscription was created (i.e. subscription should be the base/grain/group-by of this CTE).
- There's no filter specific to VPN, so wouldn't this also pick up other FxA-related subscriptions like Relay premium?

Reply:
picking up non-vpn subscriptions is intentional

Reply:
But by only picking a single subscription, if someone subscribes to multiple services in one flow they won't all get attributed.

And if only a single subscription should be attributed per flow then I'd think the flow's first subscription is probably the one that deserves attribution the most (what kicked off their buying spree).

Reply:
yeah, i think i had a bug here, I reworked this and `attributed_flows`, the end goal is to have a 1:1 relationship between flow_id and subscription_id, and i got caught up on fxa_uid.

Reply:
> And if only a single subscription should be attributed per flow then I'd think the flow's first subscription is probably the one that deserves attribution the most (what kicked off their buying spree).

I will bring it up with subplat eng, but choosing one subscription per flow, and choosing the last subscription, are both intentional.